okhttp-tests/src/test/java/okhttp3/internal/http2/HttpOverH2CTest.java

```diff
@@ -0,0 +1,85 @@
+package okhttp3.internal.http2;
+
+import okhttp3.OkHttpClient;
+import okhttp3.Protocol;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.List;
+
+import static okhttp3.TestUtil.defaultClient;
+import static org.junit.Assert.assertEquals;
+
+/**
+ * Test H2C, the cleartext implementation of HTTP/2.
+ *
+ * @author jpitz
+ */
+@SuppressWarnings("Duplicates")
+public final class HttpOverH2CTest extends Http2TestBase {
+  private static final List<Protocol> PROTOCOLS = Arrays.asList(Protocol.H2C);
+
+  @Before
+  public void setUp() throws Exception {
+    super.setUp(
```

Comment:
It feels like this could be handled without as much major surgery with a @Parameterized test.  Not sure which is generally favoured, but I try to avoid subclassing in case there is another dimension of freedom to the tests in the future e.g. Platform.

But interested to hear input from others and yours.

Reply:
It also means you can't go to a single test and run it from within Http2TestBase.

Reply:
Huh.. This week I learned. Didn't know about Parameterized builds in JUnit. This is exactly what I am looking for.